galaxy/templates/cronjob-maintenance.yaml

```diff
@@ -1,47 +1,102 @@
-            imagePullPolicy: {{ .Values.image.pullPolicy }}
-            # delete all tmp files older than walltime limit
+          - name: galaxy-cron-{{ $key }}
+            image: "{{ $.Values.image.repository }}:{{ $.Values.image.tag }}"
+            imagePullPolicy: {{ $.Values.image.pullPolicy }}
+            {{- if or $cronjob.defaultEnv $cronjob.extraEnv }}
+            env:
+            {{- if $cronjob.defaultEnv }}
+            {{- include "galaxy.podEnvVars" $}}
+            {{- end }}
+            {{- if $cronjob.extraEnv }}
+            {{- $cronjob.extraEnv | toYaml | nindent 12 }}
+            {{- end }}
+            {{- end }}
             command:
-              - find
-              - {{ .Values.persistence.mountPath }}/tmp
-              - '!'
-              - -newermt
-              - -{{ (index .Values "configs" "job_conf.yml" "runners" "k8s" "k8s_walltime_limit" | default 604800) }} seconds
-              - -type
-              - f
-              - -exec
-              - rm
-              - '{}'
-              - ;
+            {{- range $cmd := $cronjob.command }}
+              - {{ tpl $cmd $ | quote }}
+            {{- end}}
+            {{- if $cronjob.args }}
+            args:
+              {{- range $arg := $cronjob.args }}
+              - {{ tpl $arg $ | quote }}
+              {{- end }}
+            {{- end }}
             volumeMounts:
             - name: galaxy-data
-              mountPath: {{ .Values.persistence.mountPath }}
+              mountPath: {{ $.Values.persistence.mountPath }}
+            {{- range $key, $entry := $cronjob.extraFileMappings }}
+            - name: {{ include "galaxy.getExtraFilesUniqueName" $key }}
+              mountPath: {{ $key }}
+              subPath: {{ include "galaxy.getFilenameFromPath" $key }}
+            {{- end }}
           volumes:
           - name: galaxy-data
-            {{- if .Values.persistence.enabled }}
+            {{- if $.Values.persistence.enabled }}
             persistentVolumeClaim:
-              claimName: {{ template "galaxy.pvcname" . }}
+              claimName: {{ template "galaxy.pvcname" $ }}
             {{- else }}
             emptyDir: {}
             {{- end }}
+          {{- range $key, $entry := $cronjob.extraFileMappings }}
+          - name: {{ include "galaxy.getExtraFilesUniqueName" $key }}
+            {{- if $entry.useSecret }}
+            secret:
+              secretName: {{ printf "%s-%s" (include "galaxy.fullname" $) (include "galaxy.getExtraFilesUniqueName" $key) }}
+            {{- else }}
+            configMap:
+              name: {{ printf "%s-%s" (include "galaxy.fullname" $) (include "galaxy.getExtraFilesUniqueName" $key) }}
+            {{- end }}
+          {{- end }}
           restartPolicy: OnFailure
+{{- if $cronjob.extraFileMappings }}
+{{- range $name, $entry := $cronjob.extraFileMappings }}
+---
+apiVersion: v1
+metadata:
+  # Extract the filename portion only
```

Comment:
An example of extrafile mappings are not shown in values.yaml. Maybe add a comment?

Reply:
Done.  I've added and example cron job and documented in the README.